testing/kokoro/system_tests.sh

```diff
@@ -54,15 +54,16 @@ SIGNIFICANT_CHANGES=$(git --no-pager diff --name-only main..HEAD | grep -Ev '(\.
 # but weren't deleted by the current PR.
 # CHANGED_DIRS will be empty when run on main.
 CHANGED_DIRS=$(echo "$SIGNIFICANT_CHANGES" | tr ' ' '\n' | grep "/" | cut -d/ -f1 | sort -u | tr '\n' ' ' | xargs --no-run-if-empty ls -d 2>/dev/null || true)
+GO_CHANGED_PKGS=$(echo "$SIGNIFICANT_CHANGES" | tr ' ' '\n' | grep "/" | tr '\n' ' ' | xargs --no-run-if-empty ls -d 2>/dev/null || true)
 
 # List all modules in changed directories.
 # If running on main will collect all modules in the repo, including the root module.
 # shellcheck disable=SC2086
-GO_CHANGED_MODULES="$(find ${CHANGED_DIRS:-.} -name go.mod)"
-# If we didn't find any modules, use the root module.
-GO_CHANGED_MODULES=${GO_CHANGED_MODULES:-./go.mod}
-# Exclude the root module, if present, from the list of sub-modules.
-GO_CHANGED_SUBMODULES=${GO_CHANGED_MODULES#./go.mod}
+GO_CHANGED_MODULES="$(find ${GO_CHANGED_PKGS:-.} -name go.mod | xargs --no-run-if-empty dirname)"
```

Comment:
I tried to execute the steps but encountered these problems:

- `GO_CHANGED_PKGS` would contain a list of go files, not a list of folders
- `GO_CHANGED_MODULES` does not contain the modified modules

Reply:
Maybe `dirname` could help:
```
$ dirname vertexai/chat/chat.go
vertexai/chat
```

Reply:
Thanks for taking a look at this!

I've added dirname to this pipeline, and to make future changes easier, i've allowed us to set $GIT_CHANGES locally to enable better testability here.

So running `GIT_CHANGES="functions/functionsv2/firebase/hellofirestore/go.mod compute/instances/foo.go" sh testing/kokoro/system_tests.sh` will show (among other things): `GO_CHANGED_PKGS='compute/instances
functions/functionsv2/firebase/hellofirestore'`

the script stops when it tries to read the nonexistant kokoro credentials, but this is still an improvement.

Reply:
That's correct. the module-based approach was never fully implemented, so these variables are informational only.